File: python/bee1021.py
valor = float(input())

notas = [100, 50, 20, 10, 5, 2]
moedas = [1, 0.50, 0.25, 0.10, 0.05, 0.01]

print('NOTAS:')
for nota in notas:
    quantidade_nota = int(round(valor, 2) / nota)
    print(f'{quantidade_nota} nota(s) de R$ {nota:.2f}')
    valor -= quantidade_nota * nota

print('MOEDAS:')
for moeda in moedas:
    quantidade_moeda = int(round(valor, 2) / moeda)
    print(f'{quantidade_moeda} moeda(s) de R$ {moeda:.2f}')
    valor -= quantidade_moeda * moeda

File: java/bee1021.java
import java.util.Scanner;

public class bee1021 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        double valor = input.nextDouble();
        input.close();

        Caixa caixa = new Caixa(valor);
        caixa.getValorTrocado();
    }
}

class Caixa {
    private double valor;

    Caixa(double valor) {
        this.valor = valor;
    }

    public void getValorTrocado() {
        int[] notas = {100, 50, 20, 10, 5, 2};
        double[] moedas = {1, 0.5, 0.25, 0.10, 0.05, 0.01};
        
        int valorInteiro = (int) valor;
        double valorRestante = valor - valorInteiro;

        System.out.println("NOTAS :");
        for(int nota : notas) {
            int qtdNotas = valorInteiro / nota;
            valorInteiro %= nota;
            System.out.printf("%d nota(s) de R$ %d.00%n", qtdNotas, nota);
        }

        int valorQuebrado = (int) Math.round(valorRestante * 100) + (valorInteiro * 100);

        System.out.println("MOEDAS: ");
        for(double moeda : moedas) {
            int qtdMoedas = (int) (valorQuebrado / (moeda * 100));
            valorQuebrado %= (int) (moeda * 100);
            System.out.printf("%d moeda(s) de R$ %.2f%n", qtdMoedas, moeda);
        }
    }
}